### PR TITLE
18803 Add amalgamation email

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#18803

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by vysakh-menon-aot** [comment]
@kzdev420 can you make sure you assign yourself as Assignees for your pr's

File: queue_services/entity-emailer/src/entity_emailer/worker.py [hunk 1]
```diff
@@ -183,6 +184,9 @@ def process_email(email_msg: dict, flask_app: Flask):  # pylint: disable=too-man
             elif etype == 'specialResolution':
                 email = special_resolution_notification.process(email_msg['email'], token)
                 send_email(email, token)
+            elif etype == 'amalgamation':
```
Comment (vysakh-menon-aot) [inline queue_services/entity-emailer/src/entity_emailer/worker.py:187]: amalgamationApplicaton

Comment (kzdev420) [reply queue_services/entity-emailer/src/entity_emailer/worker.py]: updated

File: queue_services/entity-emailer/src/entity_emailer/email_processors/amalgamation_notification.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        filing_type=filing_type
+    )
+
+    # get attachments
+    pdfs = _get_pdfs(status, token, business, filing, leg_tmz_filing_date, leg_tmz_effective_date)
+    if not pdfs:
+        return {}
+
+    # get recipients
+    recipients = get_recipients(status, filing.filing_json, token)
```
Comment (vysakh-menon-aot) [inline queue_services/entity-emailer/src/entity_emailer/email_processors/amalgamation_notification.py:168]: missing one parameter (filing_type)

Comment (vysakh-menon-aot) [reply queue_services/entity-emailer/src/entity_emailer/email_processors/amalgamation_notification.py:168]: and get_recipients will return only Registered office email address when filing is in completed status

Comment (vysakh-menon-aot) [reply queue_services/entity-emailer/src/entity_emailer/email_processors/amalgamation_notification.py:168]: Can you check with Mihai/Yui why there is a difference in IA recipients and Amalgamation recipients (when filing is COMPLETED)

Comment (kzdev420) [reply queue_services/entity-emailer/src/entity_emailer/email_processors/amalgamation_notification.py:168]: I will check with him

Comment (kzdev420) [reply queue_services/entity-emailer/src/entity_emailer/email_processors/amalgamation_notification.py]: I checked with Yui said that `Amalgamation is same with IA and the email spreadsheet has an error.`
I just updated everything

File: queue_services/entity-emailer/src/entity_emailer/email_processors/amalgamation_notification.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
+# Copyright © 2021 Province of British Columbia
```
Comment (vysakh-menon-aot) [inline queue_services/entity-emailer/src/entity_emailer/email_processors/amalgamation_notification.py:1]: 2024

File: queue_services/entity-emailer/tests/unit/email_processors/test_amalgamation_notification.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# Copyright © 2022 Province of British Columbia
```
Comment (vysakh-menon-aot) [inline queue_services/entity-emailer/tests/unit/email_processors/test_amalgamation_notification.py:1]: 2024

**Review by vysakh-menon-aot — APPROVED** [review]

**Review by JazzarKarim — APPROVED** [review]
LGTM

**Pull request merged**